nservicebus/messaging/moving-handlers.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Moving Handlers
+summary: Guidelines for choosing a way how ot move handlers between the endpoints
+reviewed: 2021-09-01
+component: Core
+isLearningPath: true
+---
+
+As the distributed systems evolve, sometimes there is a need to change which Endpoint handle which message type. During development phase such changes are relativelly straightforward, but when the system is in production additional consideration need to be given to the messages that are in-flight. 
+
+## How to move Handlers in between Endpoints
+
+To move a handler from SourceEndpoint to a DestinationEndpoint the following steps needs to be taken
```

Comment:
```suggestion
To move a handler from `SourceEndpoint` to a `DestinationEndpoint`, the following steps need to be taken:
```

Reply:
I haven't made the change throughout my suggestions, but I find SourceEndpoint and DestinationEndpoint conflated as you could read them as "the endpoint sending the message" and "the endpoint receiving the message". Wouldn't it be easier to give a real-world example here? For example, a Sales endpoint that used to handle Invoicing, and we're trying to move out the ProcessInvoiceHandler to a dedicated Invoicing endpoint.

Reply:
Yeah I struggled with the naming here. I was tempted to follow the way that is used in contracts:
To move a handler from an endpoint from now on called: `Duck` to another endpoint from now on called: `Donald` or something. But still I struggled with the names. Maybe I should go with:
 - Reciever and New receiver. 
 - Empire and New Order
 - R2D2 and C3PO

Reply:
You have my blessing for the latter 😆